### PR TITLE
pipeline.options should never be None

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Also fix a typehints error this exposed.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @charlesccychen

File: sdks/python/apache_beam/pipeline.py [hunk 1]
```diff
@@ -226,12 +225,9 @@ def apply(self, transform, pvalueish=None):
     self._current_transform().add_part(current)
     self.transforms_stack.append(current)
 
-    if self.options is not None:
-      type_options = self.options.view_as(TypeOptions)
-    else:
-      type_options = None
+    type_options = self.options.view_as(TypeOptions)
```
Comment (charlesccychen) [inline sdks/python/apache_beam/pipeline.py]: remove unnecessary newline

**Comment by charlesccychen** [comment]
Thanks!  LGTM.

File: sdks/python/apache_beam/pipeline.py (hunk 1 above)
Comment (robertwb) [inline sdks/python/apache_beam/pipeline.py]: Done.

**Comment by robertwb** [comment]
Thanks for the quick review. Will merge once the tests all finish.

**Pull request closed without merging**